tasks/RTPTask.cpp

```diff
@@ -0,0 +1,365 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include <base-logging/Logging.hpp>
+#include <gst/gstcaps.h>
+#include <set>
+
+#include "Helpers.hpp"
+#include "RTPTask.hpp"
+
+using namespace std;
+using namespace gstreamer;
+using namespace base::samples::frame;
+
+RTPTask::RTPTask(std::string const& name)
+    : RTPTaskBase(name)
+{
+}
+
+RTPTask::~RTPTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See Task.hpp for more detailed
+// documentation about them.
+
+RTPSessionStatistics RTPTask::extractRTPSessionStats(GstElement* session)
+{
+    RTPSessionStatistics session_stats;
+
+    GstStructure* gst_stats = nullptr;
+    g_object_get(session, "stats", &gst_stats, NULL);
+    LOG_INFO_S << "stats: " << gst_structure_to_string(gst_stats);
+
+    session_stats.recv_nack_count = fetchUnsignedInt(gst_stats, "recv-nack-count");
+    session_stats.rtx_drop_count = fetchUnsignedInt(gst_stats, "rtx-drop-count");
+    session_stats.sent_nack_count = fetchUnsignedInt(gst_stats, "sent-nack-count");
+
+    const GValue* gst_source_stats_value =
+        gst_structure_get_value(gst_stats, "source-stats");
+    GValueArray* gst_source_stats =
+        static_cast<GValueArray*>(g_value_get_boxed(gst_source_stats_value));
+
+    if (!gst_stats) {
+        LOG_ERROR_S << "source-stats does not seem to be an array";
```

Comment:
The only thing you can infer is that it is not a boxed value. You know nothing of the array-ness of it.